app/src/main/java/com/bpn/trainme/presentation/features/home/HomeScreen.kt

```diff
@@ -48,50 +60,59 @@ fun HomeScreen(
             )
         }
     ) { paddingValues->
-        when(val state = uiState.value) {
-            is UiState.Idle -> Text("Readty to load")
-            is UiState.Loading -> CircularProgressIndicator()
-            is UiState.Error -> {
-                Box(
-                    modifier = Modifier.fillMaxSize().padding(paddingValues = paddingValues),
-                    contentAlignment = Alignment.Center) {
-                    Text(text = state.message)
+
+        PullToRefreshBox(
+            isRefreshing = isRefreshing,
+            state = pullToRefreshState,
+            onRefresh = {
+                scope.launch {
+                    viewModel.onPullToRefresh()
+                    exercises.refresh()
                 }
-            }
-            is UiState.Success ->
-                Box( modifier = Modifier.fillMaxSize().padding(paddingValues = paddingValues)) {
-                    ExerciseListContent(
-                        exercises = state.data.exercises ?: emptyList(),
-                        onExerciseClick = { id ->
-                            // Handle product click event
-                        }
-                    )
+            },
+            indicator = {
+                Indicator(
+                    modifier = Modifier.align(Alignment.TopCenter),
+                    isRefreshing = isRefreshing,
+                    containerColor = MaterialTheme.colorScheme.primaryContainer,
+                    color = MaterialTheme.colorScheme.onPrimaryContainer,
+                    state = pullToRefreshState
+                )
+            },
+        ){
+            LazyColumn(
+                modifier = Modifier.fillMaxSize().padding(paddingValues)
+            ) {
+                items(exercises.itemCount) { index ->
```

Comment:
Since you are using the paging lib, it would be beneficial to use key inside the `items()` function as well.